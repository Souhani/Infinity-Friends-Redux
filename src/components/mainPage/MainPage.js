import CreateRobot from "../../containers/CreateRobot";
import SearchBox from "../SearchBox";
import Scroll from "../Scroll";
import ErrorBoundry from "../ErrorBoundry";
// import CounterButton from "../CounterButton ";
import "./MainPage.css";
import filterRobots from "../FilterRobots";
import SelectAvatarsConnect from "../../containers/SelectAvatars_Connect";
import CardConnect from "../../containers/Card_Connect";

const MainPage = (props) => {
  let {
    onSearchFieldChange,
    searchField,
    onCreateRobotPage,
    createPage,
    onDeleteAllRobots,
    robots,
    isPending,
  } = props;
  /// if there is no robots
  const isCards = (robots) => {
    if (isPending) {
      return <h1>Loading...</h1>;
    }
    if (!robots.length) {
      return <h1>No Friends ðŸ˜”</h1>;
    } else {
      return <CardConnect filteredRobots={filterRobots(robots, searchField)} />;
    }
  };

  // if create state is true we show the user the page where he can create robots
  if (createPage) {
    return <CreateRobot />;
    // if create state is false we show the user the page where he can see robots
  } else if (!createPage) {
    return (
      <div>
        <h1>Infinity Friends</h1>
        {/* <CounterButton /> */}
        <SearchBox onSearchFieldChange={onSearchFieldChange} />
        <button
          className="b ph2 grow bg-gold pv1  br4 ba b--yellow pointer f6 "
          onClick={onDeleteAllRobots}
          style={{ position: "absolute", top: "2%", right: "2%" }}
        >
          Delete All
        </button>
        <div className="flex justify-center">
          <SelectAvatarsConnect />
          <button
            className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib ml2"
            onClick={onCreateRobotPage}
          >
            Create Friends
          </button>
        </div>
        <Scroll>
          <ErrorBoundry>{isCards(robots)}</ErrorBoundry>
        </Scroll>
      </div>
    );
  }
};

export default MainPage;
