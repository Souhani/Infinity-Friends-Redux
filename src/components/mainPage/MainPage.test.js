import { shallow } from "enzyme";
import MainPage from "./MainPage";
import filterRobots from "../FilterRobots";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onSearchFieldChange: jest.fn(),
    searchField: "",
    onRequestRobots: jest.fn(),
    requestedRobots: [],
    onCreateRobotPage: jest.fn(),
    createPage: false,
    onDeleteAllRobots: jest.fn(),
    onGetAllRobots: jest.fn(),
    robots: [],
    onDeleteRobot: jest.fn(),
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});
describe("test MainPage component", () => {
  it("MainPage renders without crashing", () => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders if there is robots", () => {
    const mockProps1 = {
      onSearchFieldChange: jest.fn(),
      searchField: "",
      onRequestRobots: jest.fn(),
      requestedRobots: [],
      onCreateRobotPage: jest.fn(),
      createPage: false,
      onDeleteAllRobots: jest.fn(),
      onGetAllRobots: jest.fn(),
      robots: [
        {
          id: 1,
          firstName: "ali",
          lastName: "ahmed",
        },
        {
          id: 2,
          firstName: "jhon",
          lastName: "snow",
        },
      ],
      onDeleteRobot: jest.fn(),
    };
    const wrapper1 = shallow(<MainPage {...mockProps1} />);
    expect.assertions(1);
    expect(wrapper1).toMatchSnapshot();
  });

  it("renders for create robots Page", () => {
    const mockProps2 = {
      onSearchFieldChange: jest.fn(),
      searchField: "",
      onRequestRobots: jest.fn(),
      requestedRobots: [],
      onCreateRobotPage: jest.fn(),
      createPage: true,
      onDeleteAllRobots: jest.fn(),
      onGetAllRobots: jest.fn(),
      robots: [],
      onDeleteRobot: jest.fn(),
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect.assertions(1);
    expect(wrapper2).toMatchSnapshot();
  });

  it("renders for request Robots", () => {
    const mockProps3 = {
      onSearchFieldChange: jest.fn(),
      searchField: "",
      onRequestRobots: jest.fn(),
      requestedRobots: [
        {
          id: 1,
          firstName: "ali",
          lastName: "ahmed",
        },
        {
          id: 1,
          firstName: "jhon",
          lastName: "snow",
        },
      ],
      onCreateRobotPage: jest.fn(),
      createPage: false,
      onDeleteAllRobots: jest.fn(),
      onGetAllRobots: jest.fn(),
      robots: [],
      onDeleteRobot: jest.fn(),
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect.assertions(1);
    expect(wrapper3).toMatchSnapshot();
  });
});
