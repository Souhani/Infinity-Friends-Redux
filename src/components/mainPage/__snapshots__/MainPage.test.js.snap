// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MainPage component MainPage renders without crashing 1`] = `
<div>
  <h1>
    Infinity Friends
  </h1>
  <SearchBox
    onSearchFieldChange={[MockFunction]}
  />
  <button
    className="b ph2 grow bg-gold pv1  br4 ba b--yellow pointer f6 "
    onClick={[MockFunction]}
    style={
      Object {
        "position": "absolute",
        "right": "2%",
        "top": "2%",
      }
    }
  >
    Delete All
  </button>
  <div
    className="flex justify-center"
  >
    <Connect(SelectAvatarsConnect) />
    <button
      className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib ml2"
      onClick={[MockFunction]}
    >
      Create Friends
    </button>
  </div>
  <Scroll>
    <ErrorBoundry>
      <h1>
        No Friends ðŸ˜”
      </h1>
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`test MainPage component renders for create robots Page 1`] = `<Connect(CreateRobot) />`;

exports[`test MainPage component renders for request Robots 1`] = `
<div>
  <h1>
    Infinity Friends
  </h1>
  <SearchBox
    onSearchFieldChange={[MockFunction]}
  />
  <button
    className="b ph2 grow bg-gold pv1  br4 ba b--yellow pointer f6 "
    onClick={[MockFunction]}
    style={
      Object {
        "position": "absolute",
        "right": "2%",
        "top": "2%",
      }
    }
  >
    Delete All
  </button>
  <div
    className="flex justify-center"
  >
    <Connect(SelectAvatarsConnect) />
    <button
      className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib ml2"
      onClick={[MockFunction]}
    >
      Create Friends
    </button>
  </div>
  <Scroll>
    <ErrorBoundry>
      <h1>
        No Friends ðŸ˜”
      </h1>
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`test MainPage component renders if there is robots 1`] = `
<div>
  <h1>
    Infinity Friends
  </h1>
  <SearchBox
    onSearchFieldChange={[MockFunction]}
  />
  <button
    className="b ph2 grow bg-gold pv1  br4 ba b--yellow pointer f6 "
    onClick={[MockFunction]}
    style={
      Object {
        "position": "absolute",
        "right": "2%",
        "top": "2%",
      }
    }
  >
    Delete All
  </button>
  <div
    className="flex justify-center"
  >
    <Connect(SelectAvatarsConnect) />
    <button
      className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib ml2"
      onClick={[MockFunction]}
    >
      Create Friends
    </button>
  </div>
  <Scroll>
    <ErrorBoundry>
      <Connect(CardConnect)
        filteredRobots={
          Array [
            Object {
              "firstName": "ali",
              "id": 1,
              "lastName": "ahmed",
            },
            Object {
              "firstName": "jhon",
              "id": 2,
              "lastName": "snow",
            },
          ]
        }
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
